File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        // approach: keep track of current min
        
        int min = prices[0];
        int result = 0;
        
        for (int price : prices) {
            if (price < min) {
                min = price;
            }
            
            result = Math.max(result, price - min);
        }
        
        return result;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int result = 0;
        int minimum = prices[0];
        
        for (int price : prices) {
            minimum = min(minimum, price);
            int profit = price - minimum;
            result = max(result, profit);
        }
        
        return result;
    }
};